### PR TITLE
register scheme for controller-manager

### PR DESCRIPTION
**Pull request by jiahuif**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR implements scheme registering for group `controllermanager.config.k8s.io`, so that we can use resources under this group.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/priority important-soon

### CODE REVIEW

File: staging/src/k8s.io/controller-manager/config/register.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	// defaulting and conversion init funcs are registered as well.
+	localSchemeBuilder = &SchemeBuilder
+	// AddToScheme is a global function that registers this API group & version to a scheme
+	AddToScheme = localSchemeBuilder.AddToScheme
+)
+
+// Adds the list of known types to the given scheme.
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
+		&LeaderMigrationConfiguration{},
```
Comment (cici37) [inline staging/src/k8s.io/controller-manager/config/register.go:43]: Should we also add `GenericControllerManagerConfiguration`?

Comment (jiahuif) [reply staging/src/k8s.io/controller-manager/config/register.go:43]: Actually, no. There is no meta in `GenericControllerManagerConfiguration, indicating it is actually never used directly through  scheme.

Comment (jiahuif) [reply staging/src/k8s.io/controller-manager/config/register.go:43]: I doubled checked. `GenericControllerManagerConfiguration` is used as nested only by `KubeControllerManagerConfiguration` and `CloudControllerManagerConfiguration`, the latter two are indeed registered.

Comment (cici37) [reply staging/src/k8s.io/controller-manager/config/register.go:43]: Yes you are right.

**Pull request merged**